### PR TITLE
modify kuberay operator crds in kuberay operator chart and add apiserver chart

### PR DESCRIPTION
**Pull request by scarlet25151**
## Why are these changes needed?

Add a new chart for deploy kuberay apiserver

The crds in helm chart is generated by the controller gen in ray-operator folder by running `make helm`

## Checks

- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: helm-chart/kuberay-apiserver/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{{- if .Values.ingress.create -}}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: kuberay-apiserver-ingress
+spec:
+  rules:
+    - host: api.kuberay.{{ .Values.ingress.host }}
```
Comment (Jeffwan) [inline helm-chart/kuberay-apiserver/templates/ingress.yaml:8]: is this by default included in the helm?

File: helm-chart/kuberay-apiserver/templates/role.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+kind: ClusterRole
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ .Values.name }}
+  name: {{ .Values.name }}
+rules:
+- apiGroups:
+  - ray.io
+  resources:
+  - rayclusters
```
Comment (Jeffwan) [inline helm-chart/kuberay-apiserver/templates/role.yaml:13]: RayServe and RayJob support is not added to ApiServer yet, right? So we just need Rayclusters now

**Comment by Jeffwan** [comment]
I am thinking is it better to embed kuberay-apiserver in kuberay helm (remove kuberay-operator). Or keep it a separate chart? Ideally, we should move everything to single public chart https://github.com/ray-project/kuberay/issues/260

File: helm-chart/kuberay-apiserver/templates/role.yaml (hunk 2 above)
Comment (scarlet25151) [reply helm-chart/kuberay-apiserver/templates/role.yaml:13]: Yes, this role only granted to resource rayclusters now.

File: helm-chart/kuberay-apiserver/templates/ingress.yaml (hunk 1 above)
Comment (Jeffwan) [reply helm-chart/kuberay-apiserver/templates/ingress.yaml:8]: If that's just for downstream. Let's remove it here? we can support it later in downstream repo

**Comment by scarlet25151** [comment]
Move the kuberay-operator to the subcharts of `kuberay` so we can use helm to deploy operator and apiserver at once use something like `helm install kuberay ./kuberay` without coupling two charts together.

**Comment by Jeffwan** [comment]
I feel it's better to keep `kuberay-operator` at this moment, once we finish the release, let's remove it. I know lots of user still use this folder and removing it will bring the breaking change

File: helm-chart/kuberay-operator/templates/role.yaml [hunk 3]
```diff
@@ -75,33 +108,50 @@ rules:
   - patch
   - update
 - apiGroups:
-  - networking.k8s.io
```
Comment (Jeffwan) [inline helm-chart/kuberay-operator/templates/role.yaml:78]: Looks like ingress are removed?

File: helm-chart/kuberay-operator/templates/role.yaml [hunk 4]
```diff
@@ -6,6 +6,15 @@ metadata:
 {{ include "kuberay-operator.labels" . | indent 4 }}
```
[the remark this answers is not in the document]
Comment (scarlet25151) [reply helm-chart/kuberay-operator/templates/role.yaml:6]: got it, it was generated by `make manifest` in operator, let me juste copy that.

File: helm-chart/kuberay-operator/templates/role.yaml (hunk 3 above)
Comment (scarlet25151) [reply helm-chart/kuberay-operator/templates/role.yaml:78]: I just copy from the https://github.com/ray-project/kuberay/blob/7ecdd22c5812db8754e470635b20d9ee4a6a425d/ray-operator/config/rbac/role.yaml let me double check

File: helm-chart/kuberay-operator/templates/role.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (scarlet25151) [reply helm-chart/kuberay-operator/templates/role.yaml:6]: resolve.

File: helm-chart/kuberay-operator/templates/role.yaml (hunk 3 above)
Comment (Jeffwan) [reply helm-chart/kuberay-operator/templates/role.yaml:78]: This is outdated. Used master instead

Comment (scarlet25151) [reply helm-chart/kuberay-operator/templates/role.yaml:78]: resolved.

**Review by Jeffwan — APPROVED** [review]

**Pull request merged**